pkg/yurthub/filter/servicetopology/filter.go

```diff
@@ -174,11 +184,246 @@ func (ssf *serviceTopologyFilter) SetStorageWrapper(s cachemanager.StorageWrappe
 	return nil
 }
 
-func (ssf *serviceTopologyFilter) Filter(req *http.Request, rc io.ReadCloser, stopCh <-chan struct{}) (int, io.ReadCloser, error) {
-	if ok := cache.WaitForCacheSync(stopCh, ssf.nodeSynced, ssf.serviceSynced, ssf.nodePoolSynced); !ok {
-		return 0, rc, nil
+func (stf *serviceTopologyFilter) Filter(obj runtime.Object, stopCh <-chan struct{}) runtime.Object {
+	if ok := cache.WaitForCacheSync(stopCh, stf.nodeSynced, stf.serviceSynced, stf.nodePoolSynced); !ok {
+		return obj
```

Comment:
If stopCh is closed, it seems to return the unfiltered object to the client. Is it suitable?

Reply:
if stopCh is closed, Yurthub will be exit and restart lately. so it's no problem to return a unfiltered object.